.github/workflows/nightly.yml

```diff
@@ -1,34 +1,64 @@
-name: Daily compatibility check against latest polkadot
+name: Nightly test
 
 on:
+  schedule:
+    - cron: '0 0 * * *'
   workflow_dispatch:
-    inputs:
-      gitlab_pipeline:
-        description: 'Link to failed pipeline'
-        required: true
-      gitlab_test_job_status:
-        description: 'GitLab test job status'
-        required: true
 
 jobs:
-  announce-status:
+  nightly-test:
     runs-on: ubuntu-latest
-    steps:
-      - name: Announce a status
-        run: echo  ${{ inputs.gitlab_test_job_status }}
 
-  create-an-issue:
-    if: ${{ inputs.gitlab_test_job_status == 'failed' }}
-    runs-on: ubuntu-latest
     steps:
-      - name: Checkout staking-miner sources
+      - name: Checkout repository
         uses: actions/checkout@v4
 
-      - name: Create an issue
+      - name: Free space on the runner
+        run: |
+          df -h
+          sudo apt -y autoremove --purge
+          sudo apt -y autoclean
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /opt/ghc
+          sudo rm -rf "/usr/local/share/boost"
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+          df -h
+
+      - name: Cache Rust dependencies
+        uses: swatinem/rust-cache@v2
+        with:
+          key: ${{ runner.os }}-rust-${{ hashFiles('**/Cargo.lock') }}
+          cache-on-failure: true
+
+      - name: Run nightly tests
+        run: |
+          sudo apt install -y protobuf-compiler
+          rustup target add wasm32-unknown-unknown
+          rustup component add rust-src
+          git clone -b master --depth 1 https://github.com/paritytech/polkadot-sdk.git polkadot-sdk
```

Comment:
@sergejparity 

Do think it would be easy to cache this binaries similar to what https://github.com/paritytech/subxt/blob/master/.github/workflows/build-substrate.yml  does?

We might be able to enable this in the CI instead of running it every night then...